app/models/manageiq/providers/azure/cloud_manager/event_catcher/stream.rb

```diff
@@ -0,0 +1,62 @@
+class ManageIQ::Providers::Azure::CloudManager::EventCatcher::Stream
+  include Vmdb::Logging
+  #
+  # Creates an event monitor
+  #
+  def initialize(client_id, client_key, azure_region, tenant_id)
+    @client_id         = client_id
+    @client_key        = client_key
+    @tenant_id         = tenant_id
+    @azure_region      = azure_region
+    @collecting_events = false
+  end
+
+  # Start capturing events
+  def start
+    @collecting_events = true
+  end
+
+  # Stop capturing events
+  def stop
+    @collecting_events = false
+  end
+
+  def each_batch
+    while @collecting_events
+
+      # Grab only events for the last minute if this is the first poll
+      startup_interval = (Time.current - 1.minute).httpdate
+      filter = @since ? "eventTimestamp ge #{@since}" : "eventTimestamp ge #{startup_interval}"
+      events = connection.list(filter).sort_by(&:event_timestamp)
+
+      # HACK: the Azure Insights API does not support the 'gt' (greater than relational operator)
+      # therefore we have to poll from 1 second past the timestamp of the last event to avoid
+      # gathering the same event more than once.
```

Comment:
I see you are fighting an ugly API here :(
But could this algorithm miss Events? Does this filter return _every_ event for "ge(since)" ? or is it paged and returns only the first 20?